app/services/katello/pulp3/content.rb

```diff
@@ -9,11 +9,17 @@ def create_upload(size = 0, checksum = nil, content_type = nil, repository = nil
           if checksum
             content_backend_service = SmartProxy.pulp_primary.content_service(content_type)
             if repository&.generic?
-              content_list = content_backend_service.content_api(repository.repository_type, content_type).list("sha256": checksum)
+              if content_type == 'ostree'
+                content_list = content_backend_service.content_api(repository.repository_type, content_type).list('checksum': checksum)
+              else
+                content_list = content_backend_service.content_api(repository.repository_type, content_type).list('sha256': checksum)
```

Comment:
i do see a 'checksum' argument for the content_refs_api, but not for the artifacts api